packages/inventory/src/components/filters/useTextFilter.js

```diff
@@ -0,0 +1,36 @@
+import { useState } from 'react';
+import { TEXTUAL_CHIP } from '../../shared';
+
+export const textFilterState = { textFilter: '' };
+export const TEXT_FILTER = 'TEXT_FILTER';
+export const textFilterReducer = (_state, { type, payload }) => ({
+    ...type === TEXT_FILTER && {
+        textFilter: payload
+    }
+});
+
+export const useTextFilter = ([ state, dispatch ] = [ textFilterState ]) => {
+    let [ value, setValue ] = useState('');
+    if (dispatch) {
+        value = state.textFilter;
+        setValue = (newValue) => dispatch({ type: TEXT_FILTER, payload: newValue });
```

Comment:
I would probably make a custom handler instead of overriding the `useState` callback
```jsx
handleSomething = newValue => dispatch ? dispatch(...) : setValue(...)
```